app/main.py

```diff
@@ -86,129 +89,75 @@
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
-
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
-
-    # apply potion if exist
-    if lancelot["potion"] is not None:
-        if "power" in lancelot["potion"]["effect"]:
-            lancelot["power"] += lancelot["potion"]["effect"]["power"]
-
-        if "protection" in lancelot["potion"]["effect"]:
-            lancelot["protection"] += lancelot["potion"]["effect"]["protection"]
-
-        if "hp" in lancelot["potion"]["effect"]:
-            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
-
-    # arthur
-    arthur = knightsConfig["arthur"]
+class Knight:
+    def __init__(self, data: dict) -> None:
+        self.name = data["name"]
+        self.power = data["power"]
+        self.hp = data["hp"]
+        self.armour = data["armour"]
+        self.protection = 0
+        self.weapon = data["weapon"]
+        self.potion = data.get("potion")
 
-    # apply armour
-    arthur["protection"] = 0
-    for a in arthur["armour"]:
-        arthur["protection"] += a["protection"]
+    def apply_armour(self) -> None:
+        self.protection = sum(element["protection"] for element in self.armour)
 
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
+    def apply_potion_if_exist(self) -> None:
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:
+                self.power += self.potion["effect"]["power"]
 
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
+            if "protection" in self.potion["effect"]:
+                self.protection += self.potion["effect"]["protection"]
 
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
+            if "hp" in self.potion["effect"]:
+                self.hp += self.potion["effect"]["hp"]
 
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
 
-    # mordred
-    mordred = knightsConfig["mordred"]
+    def check_fell(self) -> None:
+        if self.hp <= 0:
+            self.hp = 0
 
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
+    def results_hp(self, other_knight: Knight) -> None:
+        self.hp -= other_knight.power - self.protection
+        other_knight.hp -= self.power - other_knight.protection
 
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
+    def battle_pre(self, other_knight: Knight) -> None:
+        # BATTLE PREPARATIONS:
+        self.apply_armour()
+        self.apply_weapon()
+        self.apply_potion_if_exist()
```

Comment:
```suggestion
        for instance in (self, other_knight):
            # BATTLE PREPARATIONS:
            instance.apply_armour()
            instance.apply_weapon()
            instance.apply_potion_if_exist()
```

you can use a loop to avoid code duplication